### PR TITLE
fix(forge): update forge remove

### PR DESCRIPTION
**Pull request by DaniPopes**
## Motivation

- Calling `rm` command...

- Use `git rm path/to/submodule`. See: <https://stackoverflow.com/a/1260982>

### CODE REVIEW

File: cli/src/opts/dependency.rs [hunk 1]
```diff
@@ -123,6 +123,13 @@ impl FromStr for Dependency {
     }
 }
 
+impl Dependency {
+    /// Returns the name of the dependency, prioritizing the alias if it exists.
+    pub fn name(&self) -> &String {
+        self.alias.as_ref().unwrap_or(&self.name)
```
Comment (mattsse) [inline cli/src/opts/dependency.rs:129]:
```suggestion
    pub fn name(&self) -> &str {
        self.alias.as_deref().unwrap_or(self.name.as_str())
```

File: cli/src/cmd/forge/remove.rs [hunk 2]
```diff
@@ -28,58 +28,28 @@ impl Cmd for RemoveArgs {
...
-            Command::new("git")
-                .args(["submodule", "deinit", "-f", &dep_path.display().to_string()])
-                .current_dir(&git_root)
-                .exec()?;
-
-            // remove the submodule repository from .git/modules directory
-            Command::new("rm")
-                .args(["-rf", &git_mod_path.display().to_string()])
-                .current_dir(&git_root)
-                .exec()?;
```
Comment (mattsse) [inline cli/src/cmd/forge/remove.rs:74]: why do we no longer need this?

**Review by mattsse — CHANGES_REQUESTED** [review]

File: cli/src/cmd/forge/remove.rs (hunk 2 above)
Comment (mattsse) [reply cli/src/cmd/forge/remove.rs:74]: nvm read the linked SO post

File: cli/src/opts/dependency.rs (hunk 1 above)
Comment (DaniPopes) [reply cli/src/opts/dependency.rs:129]: Meh it's the same for return types

**Review by mattsse — APPROVED** [review]

**Review by mattsse — APPROVED** [review]

**Comment by mattsse** [comment]
@DaniPopes I reverted this because merged carelessly, this didn't pass the existing install+remove tests, sorry about this.

feel free to reopen this 

I think we need a separate job for all the repro tests because they use rpc calls and are very flaky

**Comment by DaniPopes** [comment]
I see, before we always removed the dependency, while now `git rm` won't remove it when there are uncommitted changes etc like in other commands:

image

I think it's better this way, with a `--force|-f` option like in `forge init`.

**Comment by DaniPopes** [comment]
It will also not remove the directory in `.git/modules/`, I don't know if this is necessary

**Comment by mattsse** [comment]
> It will also not remove the directory in .git/modules/, I don't know if this is necessary

I see, so this can be reapplied after tests are updated?

**Pull request merged**